src/plugin/plugin.js

```diff
@@ -237,16 +239,32 @@ class Plugin {
 
     // The second switch manages plugin level messages.
     switch (msg.messageType) {
-      case MessageType.ADAPTER_ADDED_NOTIFICATION:
+      case MessageType.ADAPTER_ADDED_NOTIFICATION: {
         adapter = new AdapterProxy(this.pluginServer.manager,
                                    adapterId,
                                    msg.data.name,
                                    msg.data.packageName,
                                    this);
         this.adapters.set(adapterId, adapter);
         this.pluginServer.addAdapter(adapter);
-        return;
 
+        // Tell the adapter about all saved things
+        const send = (thing) => {
+          adapter.sendMsg(
+            MessageType.DEVICE_SAVED_NOTIFICATION,
+            {
+              deviceId: thing.id,
+              device: thing.getDescription(),
+            }
+          );
+        };
+
+        Things.getThings().then((things) => {
+          things.forEach(send);
```

Comment:
Should this filter the things and only send the things which are associated with the adapter? This looks like every adapter will get every thing.

Reply:
Ideally, yes, but there's no way to tell when the gateway starts up which things belong to which adapter. Can you think of a way?

For instance, take this scenario:
* User adds a device, `device-1`, via some add-on which doesn't do auto-discovery (like, say, zigbee).
* User restarts gateway.
* Gateway loads `device-1` from database, but add-on never reports it, since there's no autodiscovery.

How could the add-on find out about the saved device, unless we send all devices to every adapter?

I'd also argue that it's trivial for any add-on to just read the database (or use the REST API) and get the list of things, so this doesn't really change the status quo.